framework/src/utils/MooseEnum.C

```diff
@@ -46,62 +43,56 @@ MooseEnum::withNamesFrom(const MooseEnumBase & other_enum)
 /**
  * Private constuctor for use by libmesh::Parameters
  */
-MooseEnum::MooseEnum() : _current_id(INVALID_ID) {}
+MooseEnum::MooseEnum() : _current("", INVALID_ID) {}
 
-MooseEnum::MooseEnum(const MooseEnumBase & other_enum) : MooseEnumBase(other_enum) {}
+MooseEnum::MooseEnum(const MooseEnumBase & other_enum)
+  : MooseEnumBase(other_enum), _current("", INVALID_ID)
```

Comment:
OK - I completely missed the fact that this copy constructor is taking the base class, not a reference to a `MooseEnum`. This is just plain weird, you'd get an incomplete initialization so this should probably just be removed.